service/src/main/java/bio/terra/pipelines/dependencies/rawls/RawlsService.java

```diff
@@ -94,11 +105,140 @@ public Entity upsertDataTableEntity(
                 .createEntity(entity, workspaceNamespace, workspaceName));
   }
 
+  public MethodConfiguration getCurrentMethodConfigForMethod(
+      String accessToken, String workspaceNamespace, String workspaceName, String methodName) {
+    return executionWithRetryTemplate(
+        listenerResetRetryTemplate,
+        () ->
+            rawlsClient
+                .getMethodConfigsApi(accessToken)
+                .getMethodConfiguration(
+                    workspaceNamespace, workspaceName, workspaceNamespace, methodName));
+  }
+
+  public ValidatedMethodConfiguration setMethodConfigForMethod(
+      String accessToken,
+      MethodConfiguration methodConfiguration,
+      String workspaceNamespace,
+      String workspaceName,
+      String methodName) {
+    return executionWithRetryTemplate(
+        listenerResetRetryTemplate,
+        () ->
+            rawlsClient
+                .getMethodConfigsApi(accessToken)
+                .updateMethodConfiguration(
+                    methodConfiguration,
+                    workspaceNamespace,
+                    workspaceName,
+                    workspaceNamespace,
+                    methodName));
+  }
+
   // returns true if submission is in a running state
   public static boolean submissionIsRunning(Submission submission) {
     return !FINAL_RUN_STATES.contains(submission.getStatus());
   }
 
+  /**
+   * validates a method config against the expected version, inputs, and outputs
```

Comment:
could use a little more description here - i like that you validate not only that the expected inputs and outputs are there, but also that they're pointed to the corresponding fields in the data table. if you could note that here that'd be great.